libcontainer/cgroups/fs/apply_raw.go

```diff
@@ -179,11 +180,24 @@ func (m *Manager) GetStats() (*cgroups.Stats, error) {
 }
 
 func (m *Manager) Set(container *configs.Config) error {
-	for name, path := range m.Paths {
-		sys, err := subsystems.Get(name)
-		if err == errSubsystemDoesNotExist || !cgroups.PathExists(path) {
+	for _, sys := range subsystems {
+		// We can't set this here, because after being applied, memcg doesn't
+		// allow a non-empty cgroup from having its limits changed.
+		if sys.Name() == "memory" {
```

Comment:
@crosbymichael 

**This is how this code has always worked.** The only change I've made is separating `.Set()` and `.Apply()`. The previous code **explicitly** did this this way (I actually haven't modified the `Apply()` method for `MemoryGroup`). In other words, I'm not making any change to how the `MemoryGroup` works.

And we can't "only do the pid change in this PR" because we need to `Set()` the cgroup value as late as possible in order for us to use the PIDs cgroup for all legal values. Sure, we could only do late setting for *only* the PIDs cgroup, but that's just ugly and there are other problems that this solves for other cgroups (the late setting is a common requirement for cgroups).